canopeum_backend/canopeum_backend/serializers.py

```diff
@@ -150,6 +150,9 @@ def get_admin_site_ids(self, obj: User) -> list[int]:
         return [siteadmin.site.pk for siteadmin in Siteadmin.objects.filter(user=obj)]
 
     def get_followed_site_ids(self, obj: User) -> list[int]:
+        user_role = self.get_role(obj)
+        if user_role == RoleName.MEGAADMIN:
+            return [site.pk for site in Site.objects.all()]
```

Comment:
Is this for debugging purposes or actually a feature for mega admins? How does this interact with trying to unfollow/follow sites ? (for example, is the "unfollow" button grayed out in the UI?

I don't think follow/unfollow endpoints need to be locked out though: not a security concern and it wouldn't "mess up the data", a user that get "unpromoted" from MegaAdmin would want to have kept their followed sites from before.

Reply:
This is an actual feature. From what i understood, a mega admin follows all sites by default. And yes, the sites followed by users are only correlated in the `UserSerializer`. Also, the "Follow/Unfollow" button is hidden in the UI for Mega Admins, so while a user is a Mega Admin, he cannot follow or unfollow sites.

Reply:
Awesome! I wasn't sure if this was already all properly accounted for. Looks like it is !